### PR TITLE
added the point system and card choosing functionality

### PR DESCRIPTION
**Pull request by IsabellaP698**

### CODE REVIEW

File: Assets/Scripts/MemMad.cs [hunk 1]
```diff
@@ -6,21 +6,27 @@
...
-    public static string[] values = new string[] { "A", "2", "3", "4", "5", "6", "7", "8", "9", "J", "Q", "K"};
+    public static string[] values = new string[] { "A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K"};
+
+    //cards that go into the correct positions
     public List<string>[] tops;
     public List<string>[] middles;
     public List<string>[] bottoms;
     public List<string>[] handcards;
 
+    //cards that go into above list<string>
```
Comment (oskarone21) [inline Assets/Scripts/MemMad.cs:29]: Are these going to be populated in the future? If not then this all can be simplified and remove uneccessary objects, check dotnet fiddle [here](https://dotnetfiddle.net/AZR1jq). Furthermore, for empty list initialisations, it's better to use` Enumerable.Empty<string>().ToList()` as it doesn't create new [object](https://agirlamonggeeks.com/2019/03/22/enumerable-empty-vs-new-ienumerable-whats-better/).

File: Assets/Scripts/MemMad.cs [hunk 2]
```diff
@@ -55,14 +61,15 @@ void Update()
...
+        //     print (card);
+        // }
         MemMadSort();
         StartCoroutine(MemMadDeal());
     }
 
+    //makes the deck
     public static List<string> GenerateDeck(){
 
         List<string> newDeck = new List<string>();
```
Comment (oskarone21) [inline Assets/Scripts/MemMad.cs:75]: Again `List<string> newDeck = Enumerable.Empty<string>().ToList()`

File: Assets/Scripts/MemMad.cs [hunk 3]
```diff
@@ -74,6 +81,7 @@ void Update()
         return newDeck;
     }
 
+    //pass in the deck and shuffle the cards
     void Shuffle<T>(List<T> list){
```
Comment (oskarone21) [inline Assets/Scripts/MemMad.cs:85]: add `private`

File: Assets/Scripts/MemMad.cs
Comment (oskarone21) [inline Assets/Scripts/MemMad.cs:1]: Line 157 add `private` accessor

File: Assets/Scripts/PointSystem.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+public class PointSystem : MonoBehaviour
+{
+
+    public TextMeshProUGUI scoreText;
```
Comment (oskarone21) [inline Assets/Scripts/PointSystem.cs:9]: can this be made private?

File: Assets/Scripts/UserInput.cs [hunk 5]
```diff
@@ -26,23 +30,62 @@ void Update()
             if (hit){
                 //clicked on a card
                 if (hit.collider.CompareTag("Card")){
-                    Card(hit.collider.gameObject);
+                    string parentObjectName = hit.collider.gameObject.transform.parent.gameObject.name;
+                    if(parentObjectName == "Hand0" || parentObjectName == "Hand1") {
```
Comment (oskarone21) [inline Assets/Scripts/UserInput.cs:34]: Can Hand0 and Hand1 be made private const string HAND_0 = 0 and so on?

**Review by oskarone21 — COMMENTED** [review]
Looks good, just some small changes :)

File: Assets/Scripts/UserInput.cs [hunk 6]
```diff
@@ -26,23 +30,62 @@ void Update()
             if (hit){
```
[the remark this answers is not in the document]
Comment (IsabellaP698) [reply Assets/Scripts/UserInput.cs:30]: combined

File: Assets/Scripts/MemMad.cs (hunk 2 above)
Comment (IsabellaP698) [reply Assets/Scripts/MemMad.cs:75]: changed

File: Assets/Scripts/MemMad.cs (hunk 3 above)
Comment (IsabellaP698) [reply Assets/Scripts/MemMad.cs:85]: added

File: Assets/Scripts/MemMad.cs
Comment (IsabellaP698) [reply Assets/Scripts/MemMad.cs:1]: added

File: Assets/Scripts/MemMad.cs (hunk 1 above)
Comment (IsabellaP698) [reply Assets/Scripts/MemMad.cs:29]: I tried that but when it compiled and I ran the game it didn't do just one card to each space - it dealt all the cards in the deck to each space instead

File: Assets/Scripts/PointSystem.cs (hunk 4 above)
Comment (IsabellaP698) [reply Assets/Scripts/PointSystem.cs:9]: cannot make the field private as the game needs to be able to access it and the UI needs to access the score system

File: Assets/Scripts/UserInput.cs (hunk 5 above)
Comment (IsabellaP698) [reply Assets/Scripts/UserInput.cs:34]: added

File: Assets/Scripts/UserInput.cs [hunk 7]
```diff
@@ -5,12 +5,20 @@
 public class UserInput : MonoBehaviour
 {
 
-    public GameObject slot1;
+    public GameObject selectedHandCard;
+
+    private PointSystem pointSys;
+
+    private string HAND_0 = "Hand0";
```
Comment (oskarone21) [inline Assets/Scripts/UserInput.cs:12]: Sorry, can you make these const as well and top of the page
```suggestion
    private const string HAND_0 = "Hand0";
```

**Review by oskarone21 — APPROVED** [review]
Just the const comment, otherwise happy with it. I'll try and see why the top 0 list and so on are necessary :)

**Pull request merged**